### PR TITLE
feat(Windows): add skip taskbar methods

### PR DESCRIPTION
**Pull request by amrbashir**
- [X] Tested on all platforms changed
- [X] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [X] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [X] Created or updated an example program if it would help users understand this functionality
- [X] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ And please only add new entries to the top of this list, right below the `# Unre
 
 # Unreleased
 
+- On Widnows, added `WindowExtWindows::set_skip_taskbar` and `WindowBuilderExtWindows::with_skip_taskbar`.
```
Comment (filnet) [inline CHANGELOG.md:11]: typo : `Widnows`

**Comment by msiglreith** [comment]
Hello! Thanks for the PR, sorry for the long silence on this topic.
Would it be possible to cover this feature with the standard windows styles or via a owner (we could even allow to use `EventLoop` as owner)? Both options should allow to hide the taskbar icon. I'm a bit hesitant with the current proposal as it might have side effects with other parts.

**Comment by amrbashir** [comment]
AFAIK, window styles can't achieve this behavior but I am open to suggestion of what window styles can do that.

Using `EventLoop`'s window as owner could work. I haven't tried it though so there might be undesirable effects. I will report back when I test this.

We have been using this in tao for quite some time now, around 6 or 7 months and there was no side effects observed. Even electron is using the same API.
TBH, I don't understand the hesitance in using `ITaskbarList` since it is specifically designed for this exact behavior and winit is already using its sibling `ITaskbarList2`.

**Review by msiglreith — COMMENTED** [review]
Fair enough, I was looking if there was a cheaper option to achieve this with the existing API but I'm fine with the TaskList API as well. (`WS_EX_TOOLWINDOW` seems to have some side-effects, the event loop owned window works as well but doesn't allow toggling at runtime). The only smaller issue I found was toggling with borderless will result in always showing the taskbar requiring explicit toggling of borderless to fix it again but that seems rather minor overall.

My only point open would be renaming the functions to `set_taskbar` and `with_taskbar`, dropping the `skip` - fits better to the current API imo.

**Comment by amrbashir** [comment]
> The only smaller issue I found was toggling with borderless will result in always showing the taskbar requiring explicit toggling of borderless to fix it again but that seems rather minor overall.

Never noticed this behavior before in `tao`; it might be because we are returning `0` in `WM_NCCALCSIZE` to achieve borderless.

> My only point open would be renaming the functions to `set_taskbar` and `with_taskbar`, dropping the skip - fits better to the current API imo.

The name was straight out taken from electron's `setSkipTaskbar` and gtk `set_skip_taskbar_hint`. I feel like `with_taskbar` and `set_taskbar` doesn't convey the main usage of the function.

**Review by msiglreith — APPROVED** [review]

**Pull request merged**